### PR TITLE
Adds a modifier function to simplify addition of local volumes to containerop

### PR DESCRIPTION
**Pull request by swiftdiaries**
The normal method of adding volumes to containerop involves 3 steps
1. Define Volume using k8s python client
2. Create step using containerop
3. Add volume / volume mount to containerop

Like this:
```
# Define volume using the k8s python client
from kubernetes import client as k8s_client
pvc = k8s_client.V1PersistentVolumeClaimVolumeSource(claim_name='claim')
volume = k8s_client.V1Volume(name='workflow', persistent_volume_claim=pvc)
volume_mount = k8s_client.V1VolumeMount(mount_path='/mnt/workflow/',name='workflow')

# Create container_op
def step_op(mount_path, step_name='confusion_matrix'):
    return dsl.ContainerOp(
        name = step_name,
        image = 'image_name:image_tag',
        arguments = [
            'arg1', arg_1,
        ]
    )
step_name = step_op('/mnt/workflow')

# Add to containerop
step_name.add_volume(volume)
step_name.add_volume_mount(volume_mount)
```

This is fairly verbose.

With this change, we can add volumes from the DSL and we don't have to interact with the k8s client. 
Syntax now:
1. Create step using containerop
2. Create volume with DSL
3. Add volume to step using DSL modifier function
```
# Create container_op
def step_op(mount_path, step_name='confusion_matrix'):
    return dsl.ContainerOp(
        name = step_name,
        image = 'image_name:image_tag',
        arguments = [
            'arg1', arg_1,
        ]
    )
step_name = step_op('/mnt/workflow')

# Volume definition
import kfp.onprem as onprem
local_volume = onprem.use_local_volume('claim', 'workflow', '/mnt/workflow')

# Add to step using modifier
step_name.apply(local_volume)
```

This is much more readable and enables better reuse of volumes across different steps.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/783)

### CODE REVIEW

**Comment by vicaire** [comment]
@swiftdiaries, what do you think of the proposal at https://github.com/kubeflow/pipelines/issues/801?

File: samples/notebooks/Lightweight Python components - basics.ipynb [hunk 1]
```diff
@@ -278,7 +278,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.6.4"
+   "version": "3.7.0"
```
Comment (Ark-kun) [inline samples/notebooks/Lightweight Python components - basics.ipynb]: This file changes does not seem to be related to the PR.

File: sdk/python/kfp/onprem.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+def use_local_volume(pvc_name='pipeline-claim', volume_name='pipeline', volume_mount_path='/mnt/pipeline'):
```
Comment (Ark-kun) [inline sdk/python/kfp/onprem.py]: How about `mount_pvc`.
"local_volume" has "hostPath" connotations.

File: samples/notebooks/Lightweight Python components - basics.ipynb (hunk 1 above)
Comment (swiftdiaries) [reply samples/notebooks/Lightweight Python components - basics.ipynb]: Left over from a previous PR.
Removed it.

File: sdk/python/kfp/onprem.py (hunk 2 above)
Comment (swiftdiaries) [reply sdk/python/kfp/onprem.py]: Makes sense 👍 Changed function name

**Comment by swiftdiaries** [comment]
Thanks for the review @Ark-kun !
